nilearn/decoding/decoder.py

```diff
@@ -870,6 +871,9 @@ def decision_function(self, X):
         y_pred: ndarray, shape (n_samples,)
             Predicted class label per sample.
         """
+        # for backwards compatibility
+        if not isinstance(X, type(np.array)):
```

Comment:
This is looking weird to me. Do you mean:

```python
if not isinstance(X, np.ndarray):
    ...
```

Usually, having `type` in an `isinstance` looks weird (sometimes it is because you intend to call `issubclass`).

Reply:
Yes you are right, thanks!